### DISCUSSION TITLE
Does snitch support data generators?

### DISCUSSION THREAD

**Discussion by brendan-simon-indt** [snitch-org/snitch#182] [Q&A]
Does snitch support data generators?
I find using data generators very convenient for testing many variations of input scenarios.

**Comment by cschreib** [comment]
Hello! You can check the [comparison with Catch2](https://github.com/snitch-org/snitch/blob/main/doc/comparison_catch2.md) page in the docs; unfortunately the answer is no. It's not too hard to replicate if you have your own `generate_some_values()` function:
```c++
TEST_CASE("foo") {
    for (const auto& value : generate_some_values()) {
        CAPTURE(value);
        // Test code using 'value' here...
    }
}
```

I tend to use this pattern with hard-coded lists, for example:
```c++
TEST_CASE("foo") {
    for (const auto value : {42, 55, 89, -5, 654, 2, 0}) {
        CAPTURE(value);
        // Test code using 'value' here...
    }
}
```

edit: See also [this issue](https://github.com/snitch-org/snitch/issues/59) which tracks a request to implement generators.
